src/node/jsonrpc.rs

```diff
@@ -0,0 +1,172 @@
+use crate::node::config::{IPCAgentJsonRPCNodeConfig, DEFAULT_JSON_RPC_ENDPOINT};
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use warp::http::StatusCode;
+use warp::reject::Reject;
+use warp::reply::with_status;
+use warp::{Filter, Rejection, Reply};
+
+/// The json rpc request param. It is the standard form our json-rpc and follows a structure similar
+/// to the one of the Ethereum RPC: https://ethereum.org/en/developers/docs/apis/json-rpc/#curl-examples
+#[derive(Serialize, Deserialize, Debug)]
+struct JSONRPCRequest {
```

Comment:
Yep yep, I will do this in a follow up PR, seems it can be shared and maybe placed in to `util` or `common` module or we can just import from `jsonrpc` mod?

Reply:
This goes in the line of a previous comment I made, if it is not a lot of work let's just move it right away instead of waiting for a follow-up PR. WDYT?

Reply:
The current `jsonrpc` mod contains the request and response, which was not merged when this code was written. Since those are merged, I can just use those directly.

Reply:
I am fine either way. Whatever works best for you. Whenever this happens feel free to stack a new PR with the proposed change and link it from here so we can have the overview of the future changes and so you are not blocked by this being merged as you did with the config change

Reply:
+1 for doing this in a separate PR

Reply:
BTW, it might be a good idea to address that PR first and then come back to this one.

Reply:
I think if it's not a big issue, maybe get this one merged first? Too many stacking up PRs will create quite some conflicts.

Reply:
I have create this issue to track it: #24, resolving this for now.